File: ShallowWaterSolvers/compressible/rossbywave/cacheCphi1.cpp
#include "nr.h"
#include "parameters.h"

void NR::cacheCphi1(Mat_O_DP &Cphi1, Vec_I_DP &phi)
{
	// Computes the values of the cosine phi1 basis
	// functions at each collocation point.

	for (int n=1;n<=N+1;n++)
	{
		for (int i=1;i<=N;i++)
		{
			Cphi1[n-1][i-1]=cos((2.0*n-1.0)*phi[i-1]);
		}
	}
}

File: Results/Plots/Fourier Series methods/old code/Different truncation methods/Truncation 8/cacheCphi1.m
function Cphi1 = cacheCphi1(N,phi)

%// Computes the values of the cosine basis
%// functions at each collocation point.

Cphi1=zeros(N+1,N);

for n=1:N+1
	for i=1:N
		Cphi1(n,i)=cos((2*n-1)*phi(i,1));
    end
end
